File: installer/src/main/java/com/teamscale/profiler/installer/VersionProvider.java
package com.teamscale.profiler.installer;

import picocli.CommandLine;

import java.util.ResourceBundle;

/**
 * Calculates the version string shown with {@code -V}.
 */
public class VersionProvider implements CommandLine.IVersionProvider {

	private static final String VERSION;

	static {
		ResourceBundle bundle = ResourceBundle.getBundle("com.teamscale.profiler.installer.app");
		VERSION = bundle.getString("version");
	}

	@Override
	public String[] getVersion() {
		return new String[]{"Teamscale profiler installer version " + VERSION};
	}
}

File: installer/src/main/kotlin/com/teamscale/profiler/installer/VersionProvider.kt
package com.teamscale.profiler.installer

import picocli.CommandLine
import java.util.*

/**
 * Calculates the version string shown with `-V`.
 */
class VersionProvider : CommandLine.IVersionProvider {
	override fun getVersion() = arrayOf("Teamscale profiler installer version $VERSION")

	companion object {
		private val VERSION: String

		init {
			val bundle = ResourceBundle.getBundle("com.teamscale.profiler.installer.app")
			VERSION = bundle.getString("version")
		}
	}
}